labrador/src/rq.rs

```diff
@@ -118,6 +119,68 @@ impl<const D: usize> Rq<D> {
     pub fn is_equal(&self, other: &Self) -> bool {
         self.coeffs == other.coeffs
     }
+
+    /// Generate random small polynomial for commitments
+    pub fn random_small() -> Self {
+        let mut rng = rand::rng();
```

Comment:
It would be nice to have a type-enforced requirement for rng to implement https://docs.rs/rand/latest/rand/trait.CryptoRng.html  trait in order to avoid any sudden breach because of the changes in the `rand` crate.

E.g.

```rust
let rng: &mut dyn rand::CryptoRng = &mut rand::rng();
```

Reply:
I am not sure I got your point on how it can be breached. Here is my proposed fix fed1eac

Reply:
Yes, it works! The point is that we should have a `CryptoRng` set explicitly as a trait bound. 
The point was that if in some future versions of `rand` crate `rand::rng()` may start returning a non-cryptographic rng and a blind update of dependencies would lead to a security breach.